ctapipe/reco/HillasReconstructor.py

```diff
@@ -218,30 +220,58 @@ def initialize_hillas_planes(
-            # we just need any point on the main shower axis a bit away from the cog
-            p2_x = moments.x + 0.1 * u.m * np.cos(moments.psi)
-            p2_y = moments.y + 0.1 * u.m * np.sin(moments.psi)
-            focal_length = subarray.tel[tel_id].optics.equivalent_focal_length
 
             pointing = SkyCoord(
                 alt=telescopes_pointings[tel_id].alt,
                 az=telescopes_pointings[tel_id].az,
                 frame=horizon_frame,
             )
 
-            camera_frame = CameraFrame(
-                focal_length=focal_length, telescope_pointing=pointing
-            )
+            focal_length = subarray.tel[tel_id].optics.equivalent_focal_length
 
-            cog_coord = SkyCoord(x=moments.x, y=moments.y, frame=camera_frame,)
-            cog_coord = cog_coord.transform_to(horizon_frame)
+            if moments.x.unit == u.Unit("m"):  # Image parameters are in CameraFrame
+
+                # we just need any point on the main shower axis a bit away from the cog
+                p2_x = moments.x + 0.1 * u.m * np.cos(moments.psi)
+                p2_y = moments.y + 0.1 * u.m * np.sin(moments.psi)
+
+                camera_frame = CameraFrame(
+                    focal_length=focal_length, telescope_pointing=pointing
+                )
+
+                cog_coord = SkyCoord(x=moments.x, y=moments.y, frame=camera_frame)
+                p2_coord = SkyCoord(x=p2_x, y=p2_y, frame=camera_frame)
+
+            else:  # Image parameters are already in TelescopeFrame
+
+                self.mode = "TelescopeFrame"
 
-            p2_coord = SkyCoord(x=p2_x, y=p2_y, frame=camera_frame)
+                # we just need any point on the main shower axis a bit away from the cog
+                p2_delta_alt = moments.y + 0.1 * u.deg * np.sin(moments.psi)
+                p2_delta_az = moments.x + 0.1 * u.deg * np.cos(moments.psi)
+
+                telescope_frame = TelescopeFrame(telescope_pointing=pointing)
+
+                cog_coord = SkyCoord(
+                    fov_lon=moments.x, fov_lat=moments.y, frame=telescope_frame
+                )
+                p2_coord = SkyCoord(
+                    fov_lon=p2_delta_az, fov_lat=p2_delta_alt, frame=telescope_frame
+                )
+
+            # Coordinates in the sky
+            cog_coord = cog_coord.transform_to(horizon_frame)
             p2_coord = p2_coord.transform_to(horizon_frame)
 
-            # re-project from sky to a "fake"-parallel-pointing telescope
-            # then recalculate the psi angle
+            # DIVERGENT MODE
             if self.divergent_mode:
+
+                # re-project from sky to a "fake"-parallel-pointing telescope
+                # then recalculate the psi angle
+                # In this case there is a posterior transformation to a parallel
+                # so the origin of the coordinates shouldn't matter
+                # WARNING: possibly missing unit-test!
```

Comment:
there is now a unit test that tests divergent mode I think, so you should update this comment to remove the WARNING

Reply:
Technically there isn't...the current unit-test enables divergent pointing but uses a parallel pointing simtel file.
Thus, this unit-test is testing that **in this particular case**, the two reconstructions coincide (as they should).

Once we can call the divergent simtel test file (one of those I gave you) with this unit-test it will work properly like the first figure I posted in the PR description - in fact, the new version of the test transforms the geometry inside the telescopes loop, so in this way, it is already generalized for divergent / real-case-scenarios.

Reply:
anyhow I specified better what is missing